### PR TITLE
fix: unify review details onto one canonical review surface

### LINKED ISSUE #102: Unify Review Details publication through one canonical review comment
## Problem

Review Details can still be published as a separate issue comment instead of being merged into the canonical review summary / bounded fallback comment.

Recent evidence:
- xbmc/xbmc#28218 produced a partial-review comment and a separate Review Details comment instead of one updated post.
- The max-turns bounded fallback path in `src/handlers/review.ts` was formatting the fallback comment without a `reviewOutputKey` marker, which prevented `appendReviewDetailsToSummary(...)` from finding and updating that comment.

## Goal

Route all Review Details publication through one canonical comment update path rather than maintaining separate append vs standalone logic in multiple branches.

## Proposed direction

- Introduce a shared helper that upserts the canonical review surface for each review family:
  - deterministic summary comment
  - bounded partial / fallback comment
  - continuation merge refreshes
  - clean-review approval surface where applicable
- Treat standalone Review Details comments as a true fallback only when the canonical comment cannot be found or updated.
- Preserve existing markers and idempotency semantics.
- Add regression coverage for timeout, max-turns, continuation merge, and clean-review flows.

## Why this matters

Keeping summary and details in separate posts makes the public review harder to read, breaks the “one canonical post per review surface” expectation, and complicates idempotency / audit logic.

## Notes

A minimal fix for the bounded max-turns fallback path is being landed separately. This issue is for the broader cleanup so every Review Details path uses the same canonical-comment model.

### PR DESCRIPTION
**Pull request by keithah**
## Summary
- unify Review Details publication onto one canonical visible surface per review cycle
- use the canonical issue comment for findings/timeout/max-turns/retry flows
- use the canonical approval review body for clean approval flows
- keep standalone Review Details only as explicit degraded fallback
- tighten idempotency and mixed-surface regression coverage so canonical surface selection stays truthful

## What changed
- added internal canonical review surface helpers in `src/handlers/review.ts`
- migrated issue-comment-backed Review Details refreshes onto canonical issue-comment rebuild/upsert
- migrated clean approvals so inline Review Details live in the approval review body
- fixed approval refresh to target the exact canonical review id and keep a single Review Details block/marker
- quarantined legacy standalone helper behavior behind degraded fallback only
- renamed remaining append-era terminology to canonical/degraded language

## Verification
- `bun test ./src/handlers/review.test.ts`
- `bun test ./src/handlers/review-idempotency.test.ts`
- `bun test --max-concurrency=2 src/handlers src/lib`
  - result: `1156 pass, 0 fail`

## Issue
- closes #102

### CODE REVIEW

File: src/handlers/review.ts [hunk 1]
```diff
@@ -4380,62 +4620,48 @@ export function createReviewHandler(deps: {
...
+                      preferredKind: "issue_comment",
                       reviewDetailsBlock: buildReviewDetailsBody(),
                       botHandles: [githubApp.getAppSlug(), "claude"],
+                      summaryBody: canonicalIssueComment.body,
                       requireDegradationDisclosure: authorClassification.searchEnrichment.degraded,
                       reviewBoundedness,
                       recheckCanPublish: () =>
-                        canPublishVisibleOutput("finalized Review Details append"),
+                        canPublishVisibleOutput("finalized canonical Review Details merge"),
                     });
```
[the remark this answers is not in the document]
Comment (keithah) [reply src/handlers/review.ts:4652]: Fixed on `6c4d3fb280`.

The deterministic published-output path now carries the exact canonical issue-comment surface from the idempotency-accept path into both Review Details merges instead of re-discovering by marker during refresh.

Verification:
- `bun test ./src/handlers/review.test.ts --test-name-pattern "published-output finalization updates only the canonical issue comment when mixed surfaces exist after an issue-comment idempotency accept"`
- pass

File: src/handlers/review.ts [hunk 2]
```diff
@@ -5099,30 +5331,60 @@ export function createReviewHandler(deps: {
                           }
                         : null,
                     }),
+                    botHandles: [githubApp.getAppSlug(), "claude"],
                     requireDegradationDisclosure: authorClassification.searchEnrichment.degraded,
                     reviewBoundedness,
+                    recheckCanPublish: () => canPublishVisibleOutput("timeout canonical Review Details merge"),
                   });
```
[the remark this answers is not in the document]
Comment (keithah) [reply src/handlers/review.ts:5348]: Fixed on `6c4d3fb280`.

The timeout Review Details refresh now passes the exact bounded partial comment as `canonicalSurface`, so the follow-up merge stays pinned to that comment even when older marker-matching comments exist.

Verification:
- `bun test ./src/handlers/review.test.ts --test-name-pattern "timeout canonical Review Details merge refreshes the same bounded partial comment when older marker-matching comments exist"`
- pass

**Comment by keithah** [comment]
@kodiai review

**Comment by keithah** [comment]
Follow-up on the latest Kodiai review is addressed on `db6c99cc2f`.

What changed:
- fixed retry merge to pass the checkpoint-derived canonical issue comment into `upsertCanonicalReviewSurface(...)` so it no longer drifts to a newer marker-matching comment
- fixed `mergeReviewDetailsIntoSummaryBody(...)` so new Review Details insertion happens before later unrelated `<details>` blocks instead of using the last closing tag in the whole body
- added a regression proving the `review-comment` idempotency-accept path still publishes Review Details when no canonical issue comment exists yet

Verification:
- `bun test ./src/handlers/review.test.ts --test-name-pattern "review-comment idempotency accept still publishes Review Details when no canonical issue comment exists yet|retry merge updates the same canonical comment with merged Review Details in a single upsert|merges Review Details before later unrelated details blocks in the canonical summary body"`
- `bun test ./src/handlers/review.test.ts --test-name-pattern "clean approval|auto-approval|Review Details|timeout|published canonical visible surface|retry merge"`
- `bun test ./src/handlers/review-idempotency.test.ts`

Current results on this head:
- review handler slice: `38 pass, 0 fail`
- review idempotency slice: `16 pass, 0 fail`

Notes on the Kodiai summary items:
- `review-comment` handling is now explicitly covered and passes on the current head
- telemetry degradation persistence was already still present; no code change was needed for that specific note
- the clean-approval test note was about test shape, not a live product bug

**Comment by keithah** [comment]
Small hardening follow-up on `29dbc4e808`:

- `findCanonicalReviewSurface(..., surfaceKind: "pull_review")` now prefers the newest marker-matching review body by reverse-scanning the returned review list before selecting a match.
- This does not change the main approval-refresh fix (that path is already pinned to the exact canonical review id), but it removes an avoidable ambiguity for pull-review rediscovery in edge cases.

Fresh verification after this change:
- `bun test ./src/handlers/review.test.ts --test-name-pattern "clean approval|auto-approval|Review Details|timeout|published canonical visible surface|retry merge"`
- `bun test ./src/handlers/review-idempotency.test.ts`

Current results on this head:
- review handler slice: `38 pass, 0 fail`
- review idempotency slice: `16 pass, 0 fail`
